src/coreclr/jit/registerloongarch64.h

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// clang-format off
+
+/*****************************************************************************/
+/*****************************************************************************/
+#ifndef REGDEF
+#error  Must define REGDEF macro before including this file
+#endif
+#ifndef REGALIAS
+#define REGALIAS(alias, realname)
+#endif
+
+#define RMASK(x) (1ULL << (x))
+
+/*
+REGDEF(name, rnum,       mask, sname) */
+REGDEF(R0,      0,     0x0001, "zero")
+REGDEF(RA,      1,     0x0002, "ra"  )
+REGDEF(TP,      2,     0x0004, "tp"  )
+REGDEF(SP,      3,     0x0008, "sp"  )
+REGDEF(A0,      4,     0x0010, "a0"  )
+REGDEF(A1,      5,     0x0020, "a1"  )
+REGDEF(A2,      6,     0x0040, "a2"  )
+REGDEF(A3,      7,     0x0080, "a3"  )
+REGDEF(A4,      8,     0x0100, "a4"  )
+REGDEF(A5,      9,     0x0200, "a5"  )
+REGDEF(A6,     10,     0x0400, "a6"  )
+REGDEF(A7,     11,     0x0800, "a7"  )
+REGDEF(T0,     12,     0x1000, "t0"  )
+REGDEF(T1,     13,     0x2000, "t1"  )
+REGDEF(T2,     14,     0x4000, "t2"  )
+REGDEF(T3,     15,     0x8000, "t3"  )
+REGDEF(T4,     16,    0x10000, "t4"  )
+REGDEF(T5,     17,    0x20000, "t5"  )
+REGDEF(T6,     18,    0x40000, "t6"  )
+REGDEF(T7,     19,    0x80000, "t7"  )
+REGDEF(T8,     20,   0x100000, "t8"  )
+REGDEF(X0,     21,   0x200000, "x0"  )
+REGDEF(FP,     22,   0x400000, "fp"  )
+REGDEF(S0,     23,   0x800000, "s0"  )
+REGDEF(S1,     24,  0x1000000, "s1"  )
+REGDEF(S2,     25,  0x2000000, "s2"  )
+REGDEF(S3,     26,  0x4000000, "s3"  )
+REGDEF(S4,     27,  0x8000000, "s4"  )
+REGDEF(S5,     28, 0x10000000, "s5"  )
+REGDEF(S6,     29, 0x20000000, "s6"  )
+REGDEF(S7,     30, 0x40000000, "s7"  )
+REGDEF(S8,     31, 0x80000000, "s8"  )
+
+//NOTE for LoongArch64:
+//  The `REG_R21` which alias `REG_X0` is specially reserved !!!
+//  It can be used only by manully and should be very careful!!!
+//  e.g. right now LoongArch64's backend-codegen/emit, there is usually
+//  needed a extra regitster for some case liking
```

Comment:
```suggestion
//  a need for an extra register for cases like
```

Reply:
OK, Thanks very much for your reviewing carefully.